### PR TITLE
Storybook V4 / Ship Separate UMD + ES Builds

### PR DESCRIPTION
**Pull request by mattrothenberg**
Once upon a time, when Matt was optimizing the Lodash imports in this repo, he also thought it would be a good idea to ship ES modules instead of the CommonJS modules we had been pushing up.

Turns out that wasn't a great idea.

Thus, this PR does several things:

- Adds Webpack in order to spice up our build process (building a minified/uglified UMD build).
- Bumps Storybook to V4 and tweaks the theme to be a bit more Heroku like
- Removes the `staticfiles` script, since Webpack can do this for us via the `CopyWebpackPlugin`
- Adds `BundleAnalyzerPlugin` to more accurately diagnose bloated builds. Run it via `yarn analyze`
- Uses `LodashModuleReplacementPlugin` to shake unused methods from UMD bundle
- Replaces [Moment](https://momentjs.com) with 2kb alternative, [Day.js](https://github.com/iamkun/dayjs)
- Uses `MiniCssExtractPlugin` to ship a stylesheet alongside the minified UMD bundle. This requires us to do some simple, one-time setup in `herokudata` –

```js
import '@heroku/react-hk-components/dist/umd/react-hk-components.css'
```

For context, I decided to explore building to UMD because an unintended side effect of shipping ES Modules is that Jest (on `herokudata`) has to manually transpile both `@heroku/react-hk-components` as well as any sub-dependencies (e.g., `lodash-es`). This leads to longer tests, and more convoluted test setup.

UMD Bundle
screen shot 2018-11-15 at 4 23 42 pm

In the spirt of instrumentation, maybe we could use a tool like [bundlesize](https://github.com/siddharthkp/bundlesize) to embellish our PRs with intel about how the size of the bundle has changed.

### CODE REVIEW

File: src/HKLineChartData.tsx [hunk 1]
```diff
@@ -224,8 +225,10 @@ function formatData (dataSet) {
...
-  return dataSet.map((d) => ({
-    x: moment.utc(d[0]).toDate(),
-    y: d[1].map((v) => isFinite(v) ? v : 0),
-  }))
+  return dataSet.map((d) => {
+    return {
+      x: dayjs(d[0]).toDate(),
+      y: d[1].map((v) => isFinite(v) ? v : 0),
+    }
+  })
```
Comment (uhoh-itsmaciek) [inline src/HKLineChartData.tsx]: We're using the `=> ({})` style in a bunch of other places to return objects from functions. Do you think it's too opaque?

File: src/HKLineChartData.tsx [hunk 2]
```diff
@@ -187,7 +188,7 @@ export default class HKLineChartData extends React.PureComponent<ILineChartDataP
         r={2}
       />) : null)
 
-    const timeStamp = moment(xScale.invert(hoverIndex)).format('llll')
+    const timeStamp = dayjs(xScale.invert(hoverIndex)).format('ddd, MMM D YYYY h:ss A')
```
Comment (uhoh-itsmaciek) [inline src/HKLineChartData.tsx]: `h:ss`? Is that right? That doesn't seem right...

File: test/Storyshots.test.js [hunk 3]
```diff
@@ -1,3 +1,3 @@
 import initStoryshots from '@storybook/addon-storyshots'
 
-initStoryshots()
+initStoryshots()
```
Comment (uhoh-itsmaciek) [inline test/Storyshots.test.js]: Trailing newlines are good

Comment (uhoh-itsmaciek) [reply test/Storyshots.test.js]: Actually I don't really care here except I'd just prefer less churn.

File: src/HKLineChartData.tsx (hunk 1 above)
Comment (mattrothenberg) [reply src/HKLineChartData.tsx]: Good catch. I changed the style so I could throw in a few strategic `console.log` statements but didn't revert, haha!

File: src/HKLineChartData.tsx (hunk 2 above)
Comment (mattrothenberg) [reply src/HKLineChartData.tsx]: Feel free to double-check me: https://github.com/iamkun/dayjs/blob/master/docs/en/API-reference.md#list-of-all-available-formats

File: test/Storyshots.test.js (hunk 3 above)
Comment (mattrothenberg) [reply test/Storyshots.test.js]: Fair, this is sloppy on my part.

File: src/HKLineChartData.tsx (hunk 2 above)
Comment (uhoh-itsmaciek) [reply src/HKLineChartData.tsx]: That's what I was looking at--isn't this

>h | 1-12 | The hour, 12-hour clock
and
>ss | 00-59 | The second, 2-digits

? Why hours and seconds?

Comment (mattrothenberg) [reply src/HKLineChartData.tsx]: Eek, sorry, I misread both the docs and your comment. Will update.

File: .storybook/config.js [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import { configure, addDecorator } from '@storybook/react'
+import { withOptions } from '@storybook/addon-options'
 import React from 'react'
-
+import heroku from './heroku-theme'
```
Comment (idan) [inline .storybook/config.js]: borat_very_nice.gif

File: src/HKLineChartData.tsx [hunk 5]
```diff
@@ -225,7 +226,7 @@ function formatData (dataSet) {
     return null
   }
   return dataSet.map((d) => ({
-    x: moment.utc(d[0]).toDate(),
+    x: dayjs(d[0]).toDate(),
```
Comment (idan) [inline src/HKLineChartData.tsx]: I see that dayjs doesn't have a [`utc()`](https://momentjs.com/docs/#/parsing/utc/) method — is this actually equivalent? I mean I don't love moment's API here, but invoking means that it behaves differently when rendering, just want to make sure this isn't a regression.

**Pull request merged**